style/CONTRIBUTE.md

```diff
@@ -221,3 +221,23 @@ Shields should target 8-12px text actual-size character heights for readability:
 | <img src="../doc-img/10_px_text.svg" height=20 width=20 /> | 10px        |
 | <img src="../doc-img/8_px_text.svg" height=20 width=20 />  | 8px         |
 | <img src="../doc-img/6_px_text.svg" height=20 width=20 />  | 6px         |
+
+### Shield Definitions
+
+The `loadShields` function in style/js/shield_defs.js contains a definition object for each shield displayed on the map. A definition object can contain the following properties:
+
+- **`backgroundImage`** – A reference to the image file used as the shield background, based on the name of the file in style/icons/. To use a different image depending on the length of the inscribed text, specify an array of images.
+- **`colorLighten`** – The color to recolor the background image by.
+- **`norefImage`** – A reference to an alternative image file used when there is no `ref`. This is appropriate if some routes in the network have a `ref` tag and others do not.
+- **`notext`** – By default, a relation missing a `ref` tag will not appear as a shield. Set this property to `true` to display a shield even if it has no `ref`. This is appropriate for one-off shield networks, which are common for toll roads and touristic routes.
+- **`padding`** – An object that specifies the amount of padding on each side of the inscribed text relative to the background image.
+- **`textColor`** – The color of the inscribed text to superimpose on the background.
+- **`textLayoutConstraint`** – A strategy for constraining the text within the background image, useful for shields of certain shapes.
+
+Additionally, **`refsByWayName`** is an object mapping way names to text that can be superimposed on the background as a fallback for a missing `ref` value. (`refsByWayName` implies `notext`.) This temporary fallback is designed for use in [limited situations](https://wiki.openstreetmap.org/wiki/United_States/Unusual_highway_networks) that meet each of the following criteria:
```

Comment:
I took a pass on instructions for crafting a shield definition object. As part of these instructions, I detailed the rationale for `refsByWayName` with enough conditions that hopefully it won’t open the floodgates to any end-runs around OSM.